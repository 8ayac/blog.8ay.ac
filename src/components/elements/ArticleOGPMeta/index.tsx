import { config } from '@/site.config';
import { getLastModifiedDate } from '@/src/shared/utils';
import { Article } from '@/src/types';
import React from 'react';

export const ArticleOGPMeta: React.FC<{
  article: Article;
}> = ({ article }) => (
  <>
    <meta
      property="article:published_time"
      content={new Date(article.attributes.publishedAt).toISOString()}
    />
    <meta
      property="article:modified_time"
      content={getLastModifiedDate(article).toISOString()}
    />
    <meta property="article:author" content={config.site.maintainer} />
    {[...new Set(article.attributes.tags)].map((tag) => (
      <meta key={tag} property="article:tag" content={tag} />
    ))}
  </>
);
