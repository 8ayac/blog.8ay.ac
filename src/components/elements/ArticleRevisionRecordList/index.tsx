import { REPOSITORY_URL } from '@/src/constants/site';
import { ArticleChangeLog } from '@/src/types';
import styled from '@emotion/styled';
import React from 'react';
import urljoin from 'url-join';

const S = {
  Details: styled.details``,

  Summary: styled.summary`
    margin-bottom: 0.75rem;
    font-weight: 600;
    cursor: pointer;
    outline: none;
  `,

  LogList: styled.ul`
    padding-left: 0;
    margin: 0;
  `,

  LogListItem: styled.li`
    margin-bottom: 0.75rem;
    list-style: none;
  `,

  UpdateDateTime: styled.time``,

  UpdateDescriptionSpan: styled.span`
    font-weight: bolder;
  `,

  ToGitHubCommitAnchor: styled.a``,
};

const removeGitCommitPrefix = (commitMessage: string): string => {
  return commitMessage.replace(/^:.*?:\s+/, '');
};

const getGitCommitUrl = (commitId: string): string => {
  return urljoin(REPOSITORY_URL, 'commit', commitId);
};

const LogList: React.FC<{ log: ArticleChangeLog[] }> = ({ log }) => {
  return (
    <S.LogList>
      {log.map((record) => (
        <S.LogListItem key={record.id}>
          <S.UpdateDateTime>
            {new Date(record.date).toISOString()}
          </S.UpdateDateTime>{' '}
          <S.UpdateDescriptionSpan>
            {removeGitCommitPrefix(record.description)}
          </S.UpdateDescriptionSpan>{' '}
          (
          <S.ToGitHubCommitAnchor
            href={getGitCommitUrl(record.id)}
            target="_blank"
            rel="noopener"
          >
            #{record.id}
          </S.ToGitHubCommitAnchor>{' '}
          by {record.author})
        </S.LogListItem>
      ))}
    </S.LogList>
  );
};

export const ArticleRevisionRecordList: React.FC<{
  log: ArticleChangeLog[];
}> = ({ log }) => {
  log.sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime());

  if (log.length < 1) return <></>;

  return (
    <>
      <S.Details open>
        <S.Summary>Revision History</S.Summary>
        <LogList log={log} />
      </S.Details>
    </>
  );
};
