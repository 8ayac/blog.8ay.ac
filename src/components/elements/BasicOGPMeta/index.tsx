import { config } from '@/site.config';
import React from 'react';
import urljoin from 'url-join';

export const BasicOGPMeta: React.FC<{
  ogTitle: string;
  ogType: string;
  ogDescription: string;
  ogImageUrl?: string;
  ogUrl: string;
}> = ({ ogTitle, ogDescription, ogImageUrl, ogUrl, ogType }) => (
  <>
    <meta property="og:title" content={ogTitle} />
    <meta property="og:type" content={ogType} />
    <meta property="og:url" content={ogUrl} />
    <meta
      property="og:image"
      content={ogImageUrl || urljoin(config.site.rootUrl, 'og.png')}
    />
    <meta property="og:site_name" content={config.site.title} />
    {!!ogDescription && (
      <meta property="og:description" content={ogDescription} />
    )}
  </>
);
