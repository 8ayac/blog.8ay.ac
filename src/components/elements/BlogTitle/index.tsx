import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    user-select: none;
  `,

  TitleSpan: styled.span`
    font-size: 2.2rem;
  `,

  SubtitleSpan: styled.span`
    margin-left: 0.5rem;
    font-size: 1.4rem;
  `,
};

export const BlogTitle: React.FC<{ title: string; subtitle: string }> = ({
  title,
  subtitle,
}) => (
  <>
    <S.ComponentWrapperDiv>
      <S.TitleSpan>{title}</S.TitleSpan>
      <S.SubtitleSpan>{subtitle}</S.SubtitleSpan>
    </S.ComponentWrapperDiv>
  </>
);
