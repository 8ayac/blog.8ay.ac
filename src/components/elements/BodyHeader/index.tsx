import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import Link from 'next/link';
import React, { ReactNode } from 'react';

export const S = {
  WrapperHeader: styled.header`
    margin-bottom: 3rem;
    border-bottom: 0.25rem solid ${theme.color.border.primaryLight};
  `,

  TitleH1: styled.h1`
    margin: 0 0 1rem;
    font-size: 2.16em;
  `,

  TitleAnchor: styled.a`
    margin: 0;
    color: ${theme.color.text.primary};
    text-decoration: none;
    filter: drop-shadow(2px 2px 1px ${theme.color.green.light2});
  `,

  BeforeTitleDiv: styled.div`
    margin-bottom: 0.3rem;
  `,

  AfterTitleDiv: styled.div`
    margin-top: 0.3rem;
  `,
};

export const BodyHeader: React.FC<{
  title: string;
  linkTo?: string;
  beforeTitle?: ReactNode;
  afterTitle?: ReactNode;
}> = ({ title, linkTo, beforeTitle, afterTitle }) => (
  <>
    <S.WrapperHeader>
      {beforeTitle && <S.BeforeTitleDiv>{beforeTitle}</S.BeforeTitleDiv>}
      {linkTo ? (
        <S.TitleH1>
          <Link href={linkTo} passHref>
            <S.TitleAnchor>{title}</S.TitleAnchor>
          </Link>
        </S.TitleH1>
      ) : (
        <S.TitleH1>{title}</S.TitleH1>
      )}
      {afterTitle && <S.AfterTitleDiv>{afterTitle}</S.AfterTitleDiv>}
    </S.WrapperHeader>
  </>
);
