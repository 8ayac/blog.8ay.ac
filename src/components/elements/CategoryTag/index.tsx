import { theme } from '@/src/constants/theme';
import { getTagsPagePath } from '@/src/shared/utils';
import styled from '@emotion/styled';
import Link from 'next/link';
import React from 'react';

const S = {
  Anchor: styled.a`
    padding: 3% 1rem 4%;
    margin: 0;
    font-size: 0.8em;
    font-weight: 600;
    color: white;
    text-decoration: none;
    background-color: ${theme.color.blue.dark};
    border-radius: 3px;

    &:hover {
      color: white;
      background-color: ${theme.color.blue.base};
    }
  `,
};

export const CategoryTag: React.FC<{ name: string }> = ({ name }) => (
  <>
    <Link href={getTagsPagePath(name)} passHref>
      <S.Anchor>{name}</S.Anchor>
    </Link>
  </>
);
