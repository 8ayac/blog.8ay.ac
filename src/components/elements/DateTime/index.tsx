import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: flex;
    flex-flow: row wrap;
  `,

  DescriptionSpan: styled.span`
    margin-right: 0.5rem;
    font-weight: 600;
  `,

  ISODateTime: styled.time`
    font-weight: bolder;
    color: ${(props) => props.theme.color.dateTime.isoDateTime.text};
  `,
};

export const DateTime: React.FC<{
  date: Date | string;
  description?: string;
}> = ({ date, description }) => (
  <>
    <S.ComponentWrapperDiv>
      {description && (
        <S.DescriptionSpan>{`${description}:`}</S.DescriptionSpan>
      )}
      <S.ISODateTime>{new Date(date).toISOString()}</S.ISODateTime>
    </S.ComponentWrapperDiv>
  </>
);
