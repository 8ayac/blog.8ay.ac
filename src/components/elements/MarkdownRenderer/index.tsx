import { theme } from '@/src/constants/theme';
import styled from '@emotion/styled';
import Prism from 'prismjs';
import React, { useEffect } from 'react';
import ReactMarkdown from 'react-markdown';
import remarkGfm from 'remark-gfm';

const S = {
  ConvertedMdBody: styled.div`
    font-size: 1.6rem;
    overflow-wrap: break-word;

    h1,
    h2 {
      padding-bottom: 0.3em;
      margin-top: 2em;
      border-bottom: 1px solid ${theme.color.border.primaryLight};
    }

    h3,
    h4,
    h5,
    h6 {
      margin-top: 1em;
    }

    a {
      font-weight: 600;
    }

    code,
    pre {
      font-family: SFMono-Regular, Consolas, Liberation Mono, Menlo, monospace;
      word-break: normal;
    }

    pre {
      white-space: pre-wrap;
    }

    /* This rule is for the inline code block. */
    code:not([class]) {
      padding: 0.2em 0.4em;
      margin: 0;
      font-size: 0.85em;
      background-color: hsla(0, 0%, 92.5%);
      border-radius: 3px;
    }

    blockquote,
    details,
    dl,
    ol,
    p,
    pre,
    table,
    ul {
      margin-top: 0;
      margin-bottom: 1.8rem;
    }

    ol ol,
    ul ol {
      list-style-type: lower-roman;
    }

    li {
      > p {
        margin-top: 1.6rem;
      }

      + li {
        margin-top: 0.25em;
      }
    }

    blockquote {
      padding: 0 1em;
      margin-left: 0;
      color: ${theme.color.text.primaryLight};
      border-left: 0.25em solid ${theme.color.blue.light3};

      :first-of-type {
        margin-top: 0;
      }

      :last-of-type {
        margin-bottom: 0;
      }
    }

    /* Below is for the plugin of Prism 'Show Language' */
    div.code-toolbar > .toolbar {
      top: 0.3em;
      right: 0.3em;
      transition: none;

      span {
        font-size: 0.85em;
        border-radius: 10px;
      }
    }
  `,
};

export const MarkdownRender: React.FC<{ content: string }> = ({ content }) => {
  useEffect(() => {
    Prism.highlightAll();
  });

  return (
    <>
      <S.ConvertedMdBody>
        <ReactMarkdown source={content} plugins={[remarkGfm]} />
      </S.ConvertedMdBody>
    </>
  );
};
