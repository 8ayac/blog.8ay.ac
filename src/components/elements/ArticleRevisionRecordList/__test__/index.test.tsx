import { ArticleRevisionRecordList } from '@/src/components/elements/ArticleRevisionRecordList';
import { REPOSITORY_URL } from '@/src/constants/site';
import { theme } from '@/src/constants/theme';
import { ArticleChangeLog } from '@/src/types';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';
import urljoin from 'url-join';

describe('ArticleRevisionRecordList', () => {
  const testLog: ArticleChangeLog[] = [
    {
      id: '0000001',
      date: new Date('2021-01-02T03:04:05.000Z'),
      description: ':test_prefix: this is a test commit 01',
      author: '8ayac',
    },
    {
      id: '0000002',
      date: new Date('2021-06-07T08:09:10.000Z'),
      description: ':test_prefix: this is a test commit 02',
      author: '8ayac',
    },
    {
      id: '0000003',
      date: new Date('2021-11-12T13:14:15.000Z'),
      description: ':test_prefix: this is a test commit 03',
      author: '8ayac',
    },
  ];

  describe('is rendered correctly', () => {
    describe('to match the snapshot', () => {
      test('when the length of the log is 1 or more', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <ArticleRevisionRecordList log={testLog} />
          </ThemeProvider>
        );
        expect(wrapper).toMatchSnapshot();
      });

      test('when the length of the log is 0', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <ArticleRevisionRecordList log={[]} />
          </ThemeProvider>
        );
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('under certain conditions', () => {
      it('should not be rendered when the length of the log is 0', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <ArticleRevisionRecordList log={[]} />
          </ThemeProvider>
        );
        expect(wrapper.find('Details')).toHaveLength(0);
      });

      it('should sort the log in descending order by date', () => {
        const wrapper = mount(
          <ThemeProvider theme={theme}>
            <ArticleRevisionRecordList
              log={[testLog[0], testLog[2], testLog[1]]}
            />
          </ThemeProvider>
        );

        expect(wrapper.find('LogListItem')).toHaveLength(3);
        expect(wrapper.find('UpdateDateTime').at(0).text()).toBe(
          testLog[0].date.toISOString()
        );
        expect(wrapper.find('UpdateDateTime').at(1).text()).toBe(
          testLog[1].date.toISOString()
        );
        expect(wrapper.find('UpdateDateTime').at(2).text()).toBe(
          testLog[2].date.toISOString()
        );
      });
    });

    test('links to the detail page of the commit in GitHub properly', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleRevisionRecordList log={testLog} />
        </ThemeProvider>
      );

      expect(wrapper.find('ToGitHubCommitAnchor').at(0).prop('href')).toBe(
        urljoin(REPOSITORY_URL, 'commit', testLog[0].id)
      );
      expect(wrapper.find('ToGitHubCommitAnchor').at(1).prop('href')).toBe(
        urljoin(REPOSITORY_URL, 'commit', testLog[1].id)
      );
      expect(wrapper.find('ToGitHubCommitAnchor').at(2).prop('href')).toBe(
        urljoin(REPOSITORY_URL, 'commit', testLog[2].id)
      );
    });

    test("removes the prefix 'like `:thinking:` from descriptions", () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleRevisionRecordList log={testLog} />
        </ThemeProvider>
      );

      expect(wrapper.find('UpdateDescriptionSpan').first().text()).toBe(
        'this is a test commit 01'
      );
    });
  });

  describe('has proper style rules', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ArticleRevisionRecordList log={testLog} />
      </ThemeProvider>
    );

    test('in Summary', () => {
      expect(wrapper.find('Summary')).toHaveStyleRule('cursor', 'pointer');
      expect(wrapper.find('Summary')).toHaveStyleRule('outline', 'none');
      expect(wrapper.find('Summary')).toHaveStyleRule(
        'margin-bottom',
        '0.75rem'
      );
      expect(wrapper.find('Summary')).toHaveStyleRule('font-weight', '600');
    });

    test('in LogList', () => {
      expect(wrapper.find('LogList')).toHaveStyleRule('padding-left', '0');
      expect(wrapper.find('LogList')).toHaveStyleRule('margin', '0');
    });

    test('in LogListItem', () => {
      expect(wrapper.find('LogListItem')).toHaveStyleRule(
        'margin-bottom',
        '0.75rem'
      );
      expect(wrapper.find('LogListItem')).toHaveStyleRule('list-style', 'none');
    });

    test('in UpdateDescriptionSpan', () => {
      expect(wrapper.find('UpdateDescriptionSpan')).toHaveStyleRule(
        'font-weight',
        'bolder'
      );
    });

    test('in ToGitHubCommitAnchor', () => {
      expect(wrapper.find('ToGitHubCommitAnchor')).toHaveStyleRule(
        'font-weight',
        '600'
      );
    });
  });
});
