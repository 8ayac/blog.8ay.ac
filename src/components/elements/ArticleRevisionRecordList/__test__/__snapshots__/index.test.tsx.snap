// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleRevisionRecordList is rendered correctly to match the snapshot when the length of the log is 0 1`] = `
<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 75%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(148, 72%, 75%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 85%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 45%)",
            "bgOnHover": "hsla(212, 72%, 55%)",
            "shadow": "hsla(212, 72%, 75%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 45%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 35%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 35%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 55%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 85%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 85%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 25%)",
            "shadow": "hsla(148, 72%, 15%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 15%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 55%)",
          },
        },
      },
    }
  }
>
  <ArticleRevisionRecordList
    log={Array []}
  />
</ThemeProvider>
`;

exports[`ArticleRevisionRecordList is rendered correctly to match the snapshot when the length of the log is 1 or more 1`] = `
.emotion-2 {
  margin-bottom: 0.75rem;
  font-weight: 600;
  cursor: pointer;
  outline: none;
}

.emotion-4 {
  padding-left: 0;
  margin: 0;
}

.emotion-6 {
  margin-bottom: 0.75rem;
  list-style: none;
}

.emotion-10 {
  font-weight: bolder;
}

<ThemeProvider
  theme={
    Object {
      "color": Object {
        "articleShareButtons": Object {
          "buttonWrapperDiv": Object {
            "shadow": "hsla(212, 72%, 75%)",
          },
        },
        "blogTitle": Object {
          "logoWrapperAnchor": Object {
            "textOnHover": "hsla(148, 72%, 75%)",
          },
        },
        "bodyHeader": Object {
          "titleAnchor": Object {
            "shadow": "hsla(148, 72%, 85%)",
          },
        },
        "categoryTag": Object {
          "anchor": Object {
            "bg": "hsla(212, 72%, 45%)",
            "bgOnHover": "hsla(212, 72%, 55%)",
            "shadow": "hsla(212, 72%, 75%)",
            "text": "white",
          },
        },
        "common": Object {
          "border": Object {
            "primary": "hsla(0, 0%, 50%)",
            "primaryLight": "hsla(0, 0%, 75%)",
          },
          "text": Object {
            "accent": "hsla(148, 72%, 45%)",
            "primary": "hsla(0, 0%, 15%)",
            "primaryLight": "hsla(0, 0%, 40%)",
            "secondary": "hsla(212, 72%, 35%)",
          },
        },
        "dateTime": Object {
          "isoDateTime": Object {
            "text": "hsla(212, 72%, 35%)",
          },
        },
        "global": Object {
          "body": Object {
            "bg": "hsla(60, 72%, 55%)",
          },
          "table": Object {
            "tr": Object {
              "bg2n": "hsla(148, 72%, 85%)",
            },
          },
        },
        "markdownRenderer": Object {
          "convertedMdBody": Object {
            "blockquote": Object {
              "border": "hsla(212, 72%, 85%)",
            },
          },
        },
        "siteHeader": Object {
          "header": Object {
            "bg": "hsla(148, 72%, 25%)",
            "shadow": "hsla(148, 72%, 15%)",
          },
          "innerWrapperFlexContainerDiv": Object {
            "shadow": "hsla(148, 72%, 15%)",
          },
        },
        "socialLinkWithIcon": Object {
          "socialLinkAnchor": Object {
            "textOnHover": "hsla(60, 72%, 55%)",
          },
        },
      },
    }
  }
>
  <ArticleRevisionRecordList
    log={
      Array [
        Object {
          "author": "8ayac",
          "date": "2021-01-01T11:11:11.000Z",
          "description": ":test_prefix: this is a test commit 1-1",
          "id": "1000001",
        },
        Object {
          "author": "8ayac",
          "date": "2021-01-01T22:22:22.000Z",
          "description": ":test_prefix: this is a test commit 1-2",
          "id": "1000002",
        },
      ]
    }
  >
    <Details
      open={true}
    >
      <details
        className="emotion-0 emotion-1"
        open={true}
      >
        <Summary>
          <summary
            className="emotion-2 emotion-3"
          >
            Revision History
          </summary>
        </Summary>
        <LogList
          log={
            Array [
              Object {
                "author": "8ayac",
                "date": "2021-01-01T11:11:11.000Z",
                "description": ":test_prefix: this is a test commit 1-1",
                "id": "1000001",
              },
              Object {
                "author": "8ayac",
                "date": "2021-01-01T22:22:22.000Z",
                "description": ":test_prefix: this is a test commit 1-2",
                "id": "1000002",
              },
            ]
          }
        >
          <LogList>
            <ul
              className="emotion-4 emotion-5"
            >
              <LogListItem
                key="1000001"
              >
                <li
                  className="emotion-6 emotion-7"
                >
                  <UpdateDateTime>
                    <time
                      className="emotion-8 emotion-9"
                    >
                      2021-01-01T11:11:11.000Z
                    </time>
                  </UpdateDateTime>
                   
                  <UpdateDescriptionSpan>
                    <span
                      className="emotion-10 emotion-11"
                    >
                      this is a test commit 1-1
                    </span>
                  </UpdateDescriptionSpan>
                   
                  (
                  <ToGitHubCommitAnchor
                    href="https://github.com/8ayac/blog.8ay.ac/commit/1000001"
                    rel="noopener"
                    target="_blank"
                  >
                    <a
                      className="emotion-12 emotion-13"
                      href="https://github.com/8ayac/blog.8ay.ac/commit/1000001"
                      rel="noopener"
                      target="_blank"
                    >
                      #
                      1000001
                    </a>
                  </ToGitHubCommitAnchor>
                   
                  by 
                  8ayac
                  )
                </li>
              </LogListItem>
              <LogListItem
                key="1000002"
              >
                <li
                  className="emotion-6 emotion-7"
                >
                  <UpdateDateTime>
                    <time
                      className="emotion-8 emotion-9"
                    >
                      2021-01-01T22:22:22.000Z
                    </time>
                  </UpdateDateTime>
                   
                  <UpdateDescriptionSpan>
                    <span
                      className="emotion-10 emotion-11"
                    >
                      this is a test commit 1-2
                    </span>
                  </UpdateDescriptionSpan>
                   
                  (
                  <ToGitHubCommitAnchor
                    href="https://github.com/8ayac/blog.8ay.ac/commit/1000002"
                    rel="noopener"
                    target="_blank"
                  >
                    <a
                      className="emotion-12 emotion-13"
                      href="https://github.com/8ayac/blog.8ay.ac/commit/1000002"
                      rel="noopener"
                      target="_blank"
                    >
                      #
                      1000002
                    </a>
                  </ToGitHubCommitAnchor>
                   
                  by 
                  8ayac
                  )
                </li>
              </LogListItem>
            </ul>
          </LogList>
        </LogList>
      </details>
    </Details>
  </ArticleRevisionRecordList>
</ThemeProvider>
`;
